azurelinuxagent/ga/exthandlers.py

```diff
@@ -246,8 +246,10 @@ def run(self):
             logger.verbose(msg)
             # Log status report success on new config
             self.log_report = True
-            self.handle_ext_handlers(etag)
-            self.last_etag = etag
+
+            if self.extension_processing_allowed():
+                self.handle_ext_handlers(etag)
+                self.last_etag = etag
```

Comment:
previously we were always updating etag, now it is done only when extension processing is allowed. Is this intentional?

Reply:
This is what was breaking the previous case, we were waiting for a new GS when we technically shouldn't have processed the previous one if the extension processing was on hold.

Reply:
We should only process a GS (and update the etag) only when the extension processing is allowed

Reply:
got it, i'd rename it to "last_processed_etag" just to make it a little clearer (I was thinking this meant the last etag we received in a goal state). thanks!